docs/pipeline-configuration-files/provisioning.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: 'Provisioning'
+sidebar_position: 1
+---
+
+## Pipelines Directory
+
+When Conduit starts, it will search for all files ending with `.yml` or `.yaml` in folder `./pipelines` and its
+sub-folders. It will parse the configuration files and provision the pipelines defined in them.
+
+If you want Conduit to search a different directory, or want to provision a single file, you can specify the CLI flag
+`pipelines.path` and point to your file or directory:
+
+```shell
+./conduit -pipelines.path /path/to/my-directory
+```
+
+If the directory does not exist, Conduit will fail with the error `"pipelines.path" config value is invalid`.
+
+## Ignored errors
+
+Conduit will try to provision all pipelines regardless of the errors it may encounter while going through the
+configuration files. Invalid configuration files will be ignored (e.g. files that miss a mandatory field or have an
+invalid configuration value) and won't impact the provisioning of valid configuration files. Conduit will log a warning
+for every error it encounters during this process.
+
+You need to be careful when choosing IDs for your pipelines, connectors and processors. Each pipeline needs to get a
+unique pipeline ID, otherwise all pipelines with the same ID will be ignored. If a pipeline contains connectors or
+processors with the same ID, they will be ignored as well.
+
+**_Note_**: Connector IDs and processor IDs will be prefixed with the parent ID. In case of a connector this means the
+pipeline ID will be attached as a prefix to the connector ID (e.g. `pipelineID:connectorID`). Similarly, a pipeline
+processor's ID will be prefixed with the pipeline ID (e.g. `pipelineID:processorID`) and a connector processor's ID will
+be prefixed with the connector ID (e.g. `pipelineID:connectorID:processorID`). This ensures that connector and processor
+IDs are globally unique as long as the pipeline ID is unique.
+
+## Pipelines Immutability
+
+Pipelines provisioned by configuration files are **immutable**, any updates needed on a provisioned pipeline have to be
+done through the configuration file. You can still stop or start the pipeline through the UI or API.
```

Comment:
Nitpick: it feels that "immutable" is not quite technically correct, given that they are mutable, but the changes are restricted to files.

Reply:
we could say:
```suggestion
Pipelines provisioned by configuration files are **immutable** through the API, any updates needed on a provisioned pipeline have to be
done through the configuration files. You can only stop or start the pipeline through the UI or API.
```

Reply:
Right, I'll try to rephrase it 👍